tests/integration/security/authorization_test.go

```diff
@@ -1260,3 +1266,133 @@ func TestAuthorization_Audit(t *testing.T) {
 			rbacUtil.RunRBACTest(ctx, cases)
 		})
 }
+
+func configureExtensionProviders(t *testing.T, ctx framework.TestContext, ns string) {
+	// Get the existing mesh config.
+	systemNs := istio.ClaimSystemNamespaceOrFail(t, ctx)
+	configMap, err := ctx.Clusters().Default().CoreV1().ConfigMaps(systemNs.Name()).Get(context.Background(), "istio", v1.GetOptions{})
+	if err != nil {
+		ctx.Fatalf("Failed to get config map: %v", err)
+	}
+	meshConfig, err := mesh.ReadConfigMap(configMap, "mesh")
+	if err != nil {
+		ctx.Fatalf("Failed to read config map: %v", err)
+	}
+
+	// Create two extension providers using the same ext-authz service, one for HTTP and the other for gRPC ext-authz API.
+	service := fmt.Sprintf("%s/ext-authz.%s.svc.cluster.local", ns, ns)
+	httpProvider := &v1alpha1.MeshConfig_ExtensionProvider{
+		Name: "ext-authz-http",
+		Provider: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExtAuthzHttp{
+			EnvoyExtAuthzHttp: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExternalAuthorizationHttpProvider{
+				Service:               service,
+				Port:                  8000,
+				PathPrefix:            "/check",
+				IncludeHeadersInCheck: []string{"x-ext-authz"},
+			},
+		},
+	}
+	grpcProvider := &v1alpha1.MeshConfig_ExtensionProvider{
+		Name: "ext-authz-grpc",
+		Provider: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExtAuthzGrpc{
+			EnvoyExtAuthzGrpc: &v1alpha1.MeshConfig_ExtensionProvider_EnvoyExternalAuthorizationGrpcProvider{
+				Service: service,
+				Port:    9000,
+			},
+		},
+	}
+
+	// Update the mesh config.
+	meshConfig.ExtensionProviders = []*v1alpha1.MeshConfig_ExtensionProvider{httpProvider, grpcProvider}
+	configMap.Data["mesh"], err = protomarshal.ToYAML(meshConfig)
+	if err != nil {
+		ctx.Fatalf("Failed to marshal mesh config: %v", err)
+	}
+	if _, err := ctx.Clusters().Default().CoreV1().ConfigMaps(systemNs.Name()).Update(
+		context.Background(), configMap, v1.UpdateOptions{}); err != nil {
+		ctx.Fatalf("Failed to update config map: %v", err)
+	}
+
+	// Restart istiod to make sure the change to istiod is ready for the following tests.
+	err = ctx.Clusters().Default().CoreV1().Pods(systemNs.Name()).DeleteCollection(
```

Comment:
I don't think this is something we want to do, as we start testing multicluster and external istiod relying on mutating the Istiod deployment directly should be avoided.

cc @nmittler

Reply:
I can deploy the ext-authz service in TestMain but that means it will be there for all tests, or I can add a sleep here to wait a few seconds. I'm fine with either way, any preference?

Reply:
Actually I think we can only create the namespace ahead of time and deploy the ext-authz server in the test.

Updated to configure the mesh config before test so that we don't need to restart istiod anymore.

Reply:
the sleep would be potentionally pretty long until we add configmap watcher (we are adding this, its just not enabled yet), so maybe at the start of the test? it has no impact in mesh config until ext authz API is configured anyways right?

Reply:
I think we don't need sleep as I just put put the ext authz provider in mesh config from the very beginning. And yes the mesh config change has no impact on anything else until authz API (CUSTOM) is used on a workload.

Reply:
I wrote my comment before i saw yours/the code - yep this looks great now.

Reply:
thanks for the review!